### PR TITLE
feat: percentage based fee handler

### PR DESCRIPTION
**Pull request by nmlinaric**
## Description
Introduces new fee handler that charges % from the transferred token amount.

## Related Issue Or Context
Closes: #188 

## How Has This Been Tested? Testing details.
 - unit tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: contracts/handlers/fee/PercentageERC20FeeHandler.sol [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.11;
+
+import "../../interfaces/IBridge.sol";
+import "../../interfaces/IERCHandler.sol";
+import "../../ERC20Safe.sol";
+import { BasicFeeHandler } from "./BasicFeeHandler.sol";
+
+/**
+    @title Handles deposit fees on Substrate based on Effective rates provided by Fee oracle.
```
Comment (mpetrun5) [inline contracts/handlers/fee/PercentageERC20FeeHandler.sol:11]: I don't think it is for substrate or by the fee oracle.

File: contracts/handlers/fee/PercentageERC20FeeHandler.sol [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    @author ChainSafe Systems.
+    @notice This contract is intended to be used with the Bridge contract.
+ */
+contract PercentageFeeHandler is BasicFeeHandler, ERC20Safe {
+
+    /**
+        @notice _fee inherited from BasicFeeHandler in this implementation is
+        in BPS and should be multiplied by 10000 to avoid precision loss
+     */
+    uint256 public _lowerBound; // min fee in token amount
```
Comment (mpetrun5) [inline contracts/handlers/fee/PercentageERC20FeeHandler.sol:21]: Should this be per token @MakMuftic ?

File: test/handlers/fee/percentage/calculateFee.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    let res = await FeeHandlerRouterInstance.calculateFee.call(
+      relayer,
+      originDomainID,
+      destinationDomainID,
+      resourceID,
+      depositData,
+      feeData
+    );
+
+    assert.equal(res[0].toString(),"0");
```
Comment (lastperson) [inline test/handlers/fee/percentage/calculateFee.js:85]:
```suggestion
    assert.equal(res[0].toString(), "0");
```

File: test/handlers/fee/percentage/calculateFee.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    await PercentageFeeHandlerInstance.changeFeeBounds(resourceID, 100, 300000);
+    res = await FeeHandlerRouterInstance.calculateFee.call(
+      relayer,
+      originDomainID,
+      destinationDomainID,
+      resourceID,
+      depositData,
+      feeData
+    );
+    assert.equal(res[0].toString(),"10000");
```
Comment (lastperson) [inline test/handlers/fee/percentage/calculateFee.js:97]:
```suggestion
    assert.equal(res[0].toString(), "10000");
```

**Review by lastperson — CHANGES_REQUESTED** [review]

**Review by lastperson — APPROVED** [review]

**Review by MakMuftic — APPROVED** [review]

**Pull request merged**